lib/Backends/Interpreter/InterpreterDeviceManager.h

```diff
@@ -40,9 +40,9 @@ class InterpreterDeviceManager : public QueueBackedDeviceManager {
   const uint64_t functionCost_{1};
 
 public:
-  InterpreterDeviceManager(llvm::StringRef name = "unnamed",
+  InterpreterDeviceManager(std::unique_ptr<DeviceConfig> config = nullptr,
```

Comment:
Since we know the type of backend at this point, I wonder if all of these constructors should take in subclass pointers instead. My goal is to help quickly catch mistakes like this that might be made while writing an example or unit test:

```
auto dm = std::make_unique<InterpreterDeviceManager>(cpuDeviceConfigPtr);
```

Reply:
That still requires a check at runtime. (Ideally we'd want to dyn_cast and check). This can also be handled by the device manager that is using the config. Pushing the check down to the manager means we only need a config object defined for a manager that needs one.

Reply:
I think we will need a runtime check, but as discussed offline we don't have RTTI available.

Do you think it's worth adding a check for devices like Interpreter/CPU which probably wont use anything in the config?

Reply:
I don't think so, if we check in the device managers that actually use the config we only have to check when it is actually used. I'll add the mechanism to check kind to this PR.